.github/workflows/ci_workflow.yml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on: push
+
+env:
+  DOCKER_HUB_USERNAME: ${{ github.repository_owner }}
+  DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+  COMPOSE_FILE: docker-compose.test.yml
+  DOCKER_IMAGE: ${{ github.repository }}
+
+jobs:
+  build:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/branch-tag
+      - name: Docker login
+        run: echo "$DOCKER_HUB_TOKEN" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin
+      - name: Docker pull
+        if: env.BRANCH_TAG != 'master' && env.BRANCH_TAG != 'development'
+        run: docker-compose pull test
+      - name: Docker build
+        run: |
+          bin/docker-prepare
+          docker-compose build
+          docker-compose push test
+
+  unit_test:
+    name: Unit test
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/branch-tag
+      - name: Docker login
+        run: echo "$DOCKER_HUB_TOKEN" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin
+      - name: Docker pull
+        run: docker-compose pull
+      - name: Linters
+        run: docker-compose run test yarn lint
+      - name: RSpec
+        run: docker-compose run test bundle exec rspec --exclude-pattern "spec/systems/**/*_spec.rb" --profile
+
+  system_test:
+    name: System test
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/branch-tag
+      - name: Docker login
+        run: echo "$DOCKER_HUB_TOKEN" | docker login --username=$DOCKER_HUB_USERNAME --password-stdin
+      - name: Docker pull
+        run: docker-compose pull
+      - name: RSpec
+        run: docker-compose run test bundle exec rspec spec/systems --profile
+
+  deploy:
+    name: Invoke deployment
+    runs-on: ubuntu-latest
+    needs: [unit_test, system_test]
+    if: contains(github.ref, 'refs/heads/master') || contains(github.ref, 'refs/heads/development')
+    steps:
+      - name: Invoke deployment workflow
+        uses: benc-uk/workflow-dispatch@v1
```

Comment:
Still not sure if this is to handle deployment from both `dev` and `master` branch 🤔 Otherwise, something as done by Micky in his Elixir test application might be sufficient: https://github.com/hanam1ni/google-crawler/pull/1/files#diff-3680803a67ea02b9d6b8b783f532a8ce. Basically checking the branch.

Reply:
Because I want to trigger the deployment on both main branches after passed the test, so I added this.
Let me check with Micky about it. Not sure his implementation can run the deployment after the test

Reply:
Ping me again when you are sure what triggers the CI on both master and development branch as discussed this morning.

Reply:
I checked, `push` event is always called when we change anything on the branches, so I think we should test all the changes.
And only deploy if the changes happened on master/development.
I also split the deployment workflow into 2 files for staging and production.